pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -629,7 +629,8 @@ protected SegmentMetadataImpl extractSegmentMetadata(final String rawTableName,
           .unTarOneFile(new FileInputStream(segFile), V1Constants.MetadataKeys.METADATA_FILE_NAME);
       metadataPath = FileSystems.getDefault().getPath(baseDir, segmentNameStr + ".metadata");
       Files.copy(is, metadataPath);
-      segmentMetadata = new SegmentMetadataImpl(new File(metadataPath.toString()));
+      // TODO: crc is not loaded because it's inside creation.metadata file
```

Comment:
If the crc is not set right, then will the servers reload all segments all the time? This function worked right before. What happened? Why can't we fix it (even sub-optimally) while getting the crc correctly?